third_party/freedom-typings/freedom-common.d.ts

```diff
@@ -102,7 +113,8 @@ declare module freedom {
     // We use this specification so that you can reference freedom sub-modules by
     // an array-lookup of it's name. One day, maybe we'll have a nicer way to do
     // this.
-    [moduleName:string] : Function;
+    // TODO: explore how to use FreedomModuleFactoryManager.
+    [moduleName:string] : any;
```

Comment:
Because the actual type here is a lot more like `FreedomModuleFactoryManager` above (there is a close function). But if you give it the `FreedomModuleFactoryManager` type, then typescript has trouble resolving this type with other types in the interface. I think the best thing here is to sit down with Freedom folk and do a name-space cleanup to we have meaningful names for the various things that we can give a reasonable typing to.